Dockerfile

```diff
@@ -29,21 +29,23 @@ ENV POETRY_NO_INTERACTION=1 \
     POETRY_VIRTUALENVS_CREATE=1
 
 
-RUN pip install --upgrade pip && \
-    pip install "poetry>=2.0.0,<3.0.0"
+# Create a virtual environment for poetry and install it
+RUN python3 -m venv /poetry-venv && \
```

Comment:
you could abstract out the `/poetry-venv` repetivie part by doing what the docs say and:
```
export POETRY_HOME=/opt/poetry
python3 -m venv $POETRY_HOME
$POETRY_HOME/bin/pip install --upgrade pip
$POETRY_HOME/bin/pip install "poetry>=2.0.0,<3.0.0"
```
```